pkg/controller/dns/external_dns.go

```diff
@@ -147,14 +155,12 @@ func getLabels(instances ...instance) map[string]string {
 
 func cleanObjs(instances []instance) []cleanObj {
 	var cleanObjs []cleanObj
-	for _, instance := range instances {
-		if instance.action == clean {
-			obj := cleanObj{
-				resources: instance.resources,
-				labels:    getLabels(instance),
-			}
-			cleanObjs = append(cleanObjs, obj)
+	for _, instance := range filterAction(instances, clean) {
```

Comment:
why not just have this be 

cleanObjs = filterAction(instances, clean)?

Reply:
we need to extract the cleanobjs type from each instance, so we first need to filter to look for the clean instances, then create cleanobjs from each one.

cleanObjs = filterAction(instances, clean) would yield a list of instances, not cleanObjs